screens/Menu/CourseSchedule.js

```diff
@@ -23,53 +24,143 @@ import PropTypes from "prop-types";
-        data: ["SOEN 357 Lecture", "SOEN 357 Tutorial", "SOEN 345 Lecture"
-        ]
-    }];
+    useEffect(() => {
+        const intervalId = setInterval(() => {
+            getCalendarId();
+            getAccessToken();
+            getSwitchValue();
+        }, 1);
+        getCalendarEvents();
+        return () => clearInterval(intervalId);
+    }, [selectedCalendarId, accessToken]);
 
+    if (switchVal == "true" && calendarEventsList) {
     return (
-        <View style={styles.container}>
-            <View style={styles.menuButtonContainer}>
-                <TouchableOpacity style={styles.menuButton} onPress={goToMenu}>
-                    <Feather name="menu" style={styles.icon} />
-                </TouchableOpacity>
+            <View style={styles.container}>
+                <View style={styles.menuButtonContainer}>
+                    <TouchableOpacity style={styles.menuButton} onPress={goToMenu}>
+                        <Feather name="menu" style={styles.icon} />
+                    </TouchableOpacity>
+                </View>
+                <Text style={styles.mainLabel}>My Course Schedule</Text>
+                <View style={styles.scrollTextContainer}>
+                    {(switchVal == "true" && calendarEventsList) &&
+                        <FlatList
+                            data={calendarEventsList}
+                            keyExtractor={(item) => item.id}
+                            initialNumToRender={10}
+                            renderItem={({ item }) => (
+                                <ListItem
+                                    title={item.summary}
+                                    titleStyle={{ color: "white", paddingLeft: 20 }}
+                                    subtitle={item.description !== undefined ? item.description : "Please enter a description in your Google Calendar"}
+                                    rightTitleStyle={{ color: "white", paddingRight: 20 }}
+                                    containerStyle={{ backgroundColor: "#2A2E43" }}
+                                    subtitleStyle={{ color: "grey", paddingLeft: 20 }}
+                                    rightIcon={() => item.location &&
+                                        <CheckBox
+                                            size={30}
+                                            iconRight
+                                            iconType='material'
+                                            uncheckedIcon={'arrow-forward'}
+                                            uncheckedColor="#3ACCE1"
+                                            onPress={() => { goToDoubleSearch(item) }
+                                            }
+                                        />}
+                                />
+                            )}
+                            refreshing={refresh}
+                            onRefresh={handleRefresh}
+                        />
+                    }
```

Comment:
Right after this line, add : 
```
{calendarEventsList.NoEvent && 
   <Text>No Upcoming classes!</Text>
 }
```
You can style the Text tag like you want!

Reply:
FIXED !!